core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -333,6 +340,24 @@ private static KeyPair generateRsaKeyPair() {
     return generator.generateKeyPair();
   }
 
+  /** Sets the default string which is appended to the SQLAdmin API client User-Agent header. */
+  private static void setDefaultUserAgent(String userAgent) {
+    System.setProperty(APPLICATION_NAME_PROPERTY, userAgent);
```

Comment:
I don't think we need to store this in a system property - is only done for `getApplicationName()` for legacy reasons.